### PR TITLE
Webui tests

### LINKED ISSUE #415: Unit Tests for User Interface / React
#### My reason:

There is none unit test for React.

@ayusharma is working on it

https://github.com/ayusharma/verdaccio/tree/webui-tests

### PR DESCRIPTION
**Pull request by ayusharma**
**Type:** feature / unit test 

The following has been addressed in the PR:

*  There is a related issue
*  Unit or Functional tests are included in the PR

**Description:**

Resolves #415

### CODE REVIEW

**Comment by juanpicado** [comment]
@ayusharma we have a conflict here `jest.config.js`

**Comment by ayusharma** [comment]
resolving conflict.

File: package.json [hunk 1]
```diff
@@ -56,6 +56,7 @@
     "babel-jest": "^21.2.0",
     "babel-loader": "7.1.2",
     "babel-plugin-flow-runtime": "0.11.1",
+    "babel-plugin-syntax-async-functions": "^6.13.0",
```
Comment (juanpicado) [inline package.json]: Does not babel do it out of the box already?  `babel-plugin-syntax-async-functions` . I think after all merges we have to review the policy of presets

File: test/webui/global.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+/**
+ * Setup configuration for Jest
+ * This file includes gloabl settings for the JEST environment.
```
Comment (juanpicado) [inline test/webui/global.js]: typo --> gloabl

**Comment by juanpicado** [comment]
`transform-class-properties` plugin is not compatible with our Functional testing. If you remove that test will work again. We have a fancy https://github.com/verdaccio/verdaccio/blob/feature-jest-test/test/functional/lib/request.js class that extends from Promise and `transform-class-properties` just does not allow such behavior.  I had to deal with that weeks ago https://github.com/juanpicado/babel-promise-test

**Comment by juanpicado** [comment]
Issues:

The first one is on `.babelrc` configuration file.

* `"es2015"` preset is not needed it. `"es2015-node4"` does already include all you need. 
* `"transform-class-properties"` does break the functional test. You don't need it.

```
     "test": {
-      "presets": [ "es2015-node4", "es2015", "flow", "react"],
+      "presets": [ "es2015-node4", "flow", "react"],
       "plugins": [
         "transform-class-properties",
         "transform-object-rest-spread",
-        "transform-class-properties",
         "syntax-async-functions"
       ]
     },
```
The second issue is on `jest.config.js`.

If you map `storage` 
```
    'storage': '<rootDir>/test/webui/components/__mocks__/storage.js',
```
causes
```
    TypeError: Storage is not a constructor
```
I think you have to use something less generic than `storage`.  I'd rename all those modules as eg: `uiStorage` to avoid future conflics.

There are more issues, this is what I found by far.

**Comment by juanpicado** [comment]
If you update the `circleci.yaml` removing Node4, you are done here 👍

**Pull request merged**